e2e/test/scenarios/dashboard-cards/click-behavior.cy.spec.js

```diff
@@ -0,0 +1,692 @@
+          });
+          cy.get("aside").button("Done").click();
+
+          saveDashboard();
+
+          cy.findByTestId("dashcard").get("circle.dot").eq(POINT_INDEX).click();
+          cy.findByText("Count by Created At: Month").should("exist");
+          cy.findByTestId("qb-filters-panel").should(
+            "contain.text",
+            "Created At is April 1–30, 2026",
+          );
+          cy.findByTestId("qb-filters-panel").should(
+            "contain.text",
+            "Quantity is equal to 344",
+          );
+          cy.location().should(({ hash, pathname }) => {
+            expect(pathname).to.equal("/question");
+            const card = deserializeCardFromUrl(hash);
+            expect(card.name).to.deep.equal(TARGET_QUESTION.name);
+            expect(card.display).to.deep.equal(TARGET_QUESTION.display);
+            expect(card.dataset_query.query).to.deep.equal({
+              ...TARGET_QUESTION.query,
+              filter: ["and", QUERY_FILTER_CREATED_AT, QUERY_FILTER_QUANTITY],
+            });
+          });
+        },
+      );
+    });
+
+    it("does not allow setting saved question as custom destination if user has no permissions to it", () => {
+      cy.createCollection({ name: RESTRICTED_COLLECTION_NAME }).then(
+        ({ body: restrictedCollection }) => {
+          cy.updateCollectionGraph({
+            [USER_GROUPS.COLLECTION_GROUP]: {
+              [restrictedCollection.id]: "none",
+            },
+          });
+
+          cy.createQuestion({
+            ...TARGET_QUESTION,
+            collection_id: restrictedCollection.id,
+          });
+        },
+      );
+
+      cy.signOut();
+      cy.signInAsNormalUser();
+
+      cy.createQuestionAndDashboard({ questionDetails }).then(
+        ({ body: card }) => {
+          visitDashboard(card.dashboard_id);
+          editDashboard();
+
+          getDashboardCard().realHover().icon("click").click();
+          cy.get("aside").findByText("Go to a custom destination").click();
+          cy.get("aside").findByText("Saved question").click();
+
+          modal().findByText(RESTRICTED_COLLECTION_NAME).should("not.exist");
+          modal().findByText(TARGET_QUESTION.name).should("not.exist");
+        },
+      );
+    });
+
+    it("allows setting URL as custom destination and change it back to default click behavior", () => {
+      cy.createQuestionAndDashboard({ questionDetails }).then(
+        ({ body: card }) => {
+          visitDashboard(card.dashboard_id);
+          editDashboard();
+
+          getDashboardCard().realHover().icon("click").click();
+          cy.get("aside").findByText("Go to a custom destination").click();
+          cy.get("aside").findByText("URL").click();
+          modal().within(() => {
+            cy.findByRole("textbox").type(URL);
+            cy.button("Done").click();
+          });
+          cy.get("aside").button("Done").click();
+
+          saveDashboard();
+
+          onNextAnchorClick(anchor => {
```

Comment:
- We use [custom click behavior](https://github.com/metabase/metabase/blob/7820266e7d551b038df60b5e8317b16b7ed7a90e/frontend/src/metabase/lib/dom.js#L301-L312) which is a nightmare to test in Cypress. I tried many ways but none of them do the trick. Resources I checked/tried:
  - https://github.com/cypress-io/cypress-example-recipes/tree/master/examples/testing-dom__tab-handling-links
  - https://github.com/cypress-io/cypress-example-recipes/tree/master/examples/stubbing-spying__window
  - https://docs.cypress.io/guides/references/trade-offs#Multiple-tabs
  - https://glebbahmutov.com/blog/stub-window-open/
  - https://glebbahmutov.com/blog/cypress-tips-and-tricks/#deal-with-windowopen
  - https://stackoverflow.com/questions/64234771/cypress-prevent-open-new-tab-with-link-dynamically-created-and-clicked